clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -238,7 +269,18 @@ public synchronized void update(int requestVersion, MetadataResponse response, l
         if (isClosed())
             throw new IllegalStateException("Update requested after metadata close");
 
-        validateCluster(response.clusterId());
+        if (!validateCluster(response.clusterId())) {
+            //if validateCluster fails, do not update metadataCache with the wrong cluster information,
+            //just return and wait for next update
+            //
+            //here we don't blacklist this node from the cluster's
+            //node set since we don't have enough information from the response to map to the actual node,
+            //and since there are usually hours to days interval before we put a removed broker to a different
+            //cluster, clients should either find another node in cached node set or resolved bootstrap server
+            //again and find a new node to send update metadata request, it should be ok to not blacklist this node
+            requestUpdate();
```

Comment:
should we request a cluster metadata update here instead of regular metadata upate: e.g.,  make next call to "shouldUpdateClusterMetadata" return true.

Reply:
sure, modified.